packages/open-truss/src/configuration/engine-v1/Frame.tsx

```diff
@@ -278,6 +271,48 @@ function processProps({
   return compiledProps
 }
 
+// Processes viewProps defined by component and does the following
+// If the props is a
+// - component (e.g. <Component>) it looks up the component and sets it as the value
+// - symbol (e.g. :symbol) it looks up the symbol and sets it as the value
+// - viewProp (e.g. yaml object) it calls processViewProps recursively and sets it as the value
+// - array it calls processViewProps recursively and sets it as the value
+// - else it sets the original viewProp value
+// Returns the processed view props
+function processViewProps(
+  props: ComponentPropsShape,
+): ComponentPropsReturnShape {
+  const { configPath, viewProps, signals, COMPONENTS } = props
+  const newProps: ComponentPropsReturnShape = {}
+  for (const propName in viewProps) {
+    const prop = viewProps[propName]
+    if (isComponent(prop)) {
+      newProps[propName] = getDefaultComponent(prop, configPath, COMPONENTS)
+    } else if (isSymbol(prop)) {
+      const signalName = parseSignalName(prop) ?? ''
+      const signal = signals[signalName]
+      signal === undefined
+        ? (newProps[propName] = prop)
+        : (newProps[propName] = signal)
+    } else if (isViewProps(prop)) {
+      const subProps = Object.assign({}, props, { viewProps: prop })
+      newProps[propName] = processViewProps(subProps)
+    } else if (Array.isArray(prop)) {
+      const propsFromArray = prop.map((value) => {
+        const tmpKey = '_key'
+        const subViewProps = { viewProps: { [tmpKey]: value } }
+        const subProps = Object.assign({}, props, subViewProps)
+        return processViewProps(subProps)[tmpKey]
+      })
+
+      newProps[propName] = propsFromArray
```

Comment:
> Updates the processProps code to recursively parse through the view props and set any signals that exist in nested arrays or objects

I love me some recursion but I'm not sure this is worth it 🤔 What led you to add this? It's not very complicated so maybe it's fine to keep, but I find us handling YAML objects arrays, which it does natively, by iterating over things to be a smell.

Reply:
> but I find us handling YAML objects arrays, which it does natively

What did you mean by "it does natively" here?